examples/classification/benchmark.py

```diff
@@ -0,0 +1,393 @@
+                        are 'deepsparse', and 'onnxruntime'. Default is
+                        'deepsparse'
+  --data-path DATA_PATH
+                        path to SparseZoo stub for data sample to benchmark
+                        with, or to directory of .npz files with named sample
+                        inputs to the given model, inputs should be for a
+                        single sample with no batch dimension included.
+                        Defaults to benchmark on random image samples with
+                        channel size 3.
+  --image-shape IMAGE_SHAPE [IMAGE_SHAPE ...]
+                        Image shape to benchmark with, must be two integers.
+                        If None model is used to infer image shape
+  -b BATCH_SIZE, --batch-size BATCH_SIZE
+                        The batch size to run the benchmark for
+  -c NUM_CORES, --num-cores NUM_CORES
+                        The number of physical cores to run the benchmark on,
+                        defaults to None where it uses all physical cores
+                        available on the system. For DeepSparse benchmarks,
+                        this value is the number of cores per socket
+  -s NUM_SOCKETS, --num-sockets NUM_SOCKETS
+                        For DeepSparse benchmarks only. The number of physical
+                        cores to run the benchmark on. Defaults to None where
+                        is uses all sockets available on the system
+  -i NUM_ITERATIONS, --num-iterations NUM_ITERATIONS
+                        The number of iterations the benchmark will be run for
+  -w NUM_WARMUP_ITERATIONS, --num-warmup-iterations NUM_WARMUP_ITERATIONS
+                        The number of warmup iterations that will be executed
+                        before the actual benchmarking
+
+"""
+import argparse
+import time
+from tempfile import NamedTemporaryFile
+from typing import Any, List, Optional, Tuple, Union
+
+import numpy
+import onnx
+import onnxruntime
+from tqdm import tqdm
+
+from deepsparse import compile_model
+from deepsparse.benchmark import BenchmarkResults
+from examples.classification.helper import _BatchLoader, _load_random_data, load_data
+from sparseml.onnx.utils import (
+    get_tensor_dim_shape,
+    override_model_batch_size,
+    set_tensor_dim_shape,
+)
+from sparsezoo import Zoo
+
+
+DEEPSPARSE_ENGINE = "deepsparse"
+ORT_ENGINE = "onnxruntime"
+
+
+def _run_model(
+    args, model: Any, batch: Union[numpy.ndarray]
+) -> List[Union[numpy.ndarray]]:
+    # run model according to engine type
+    if args.engine == ORT_ENGINE:
+        outputs = model.run(
+            [out.name for out in model.get_outputs()],  # outputs
+            {model.get_inputs()[0].name: batch},  # inputs dict
+        )
+    else:  # deepsparse
+        outputs = model.run(batch)
+    return outputs
+
+
+def _get_data_loader(config, new_image_shape, total_iterations):
+    """
+    Helper function to get appropriate batch data loader
+    """
```

Comment:
nit: use inline comment for protected function

Reply:
👌